src/jit/importer.cpp

```diff
@@ -7715,7 +7715,7 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
                         unsigned   calliSlot  = lvaGrabTemp(true DEBUGARG("calli"));
                         LclVarDsc* varDsc     = &lvaTable[calliSlot];
                         varDsc->lvVerTypeInfo = tiRetVal;
-                        impAssignTempGen(calliSlot, call, clsHnd, (unsigned)CHECK_SPILL_NONE);
+                        impAssignTempGen(calliSlot, call, tiRetVal.GetClassHandle(), (unsigned)CHECK_SPILL_NONE);
```

Comment:
I think that value of clsHnd was  calliSig.retTypeClass before your fix. 

```
    if (opcode == CEE_CALLI)
    {
        /* Get the call site sig */
        eeGetSig(pResolvedToken->token, info.compScopeHnd, impTokenLookupContextHandle, &calliSig);

        callRetTyp = JITtype2varType(calliSig.retType);
        clsHnd     = calliSig.retTypeClass;
```
So it wasn't the class handle of the type that owns the method.
Am I wrong?